core/services/workflows/syncer/handler.go

```diff
@@ -390,11 +394,14 @@ func (h *eventHandler) workflowRegisteredEvent(
 	}
 
 	// Calculate the hash of the binary and config files
-	hash := workflowID(binary, config, []byte(payload.SecretsURL))
+	hash, err := pkgworkflows.GenerateWorkflowID(payload.Owner, binary, config, payload.SecretsURL)
+	if err != nil {
+		return fmt.Errorf("failed to generate workflow id: %w", err)
+	}
 
 	// Pre-check: verify that the workflowID matches; if it doesn’t abort and log an error via Beholder.
```

Comment:
is this comment wrong or is something missing? don't see anything about beholder.

Reply:
We log higher up in the thing that calls the handler